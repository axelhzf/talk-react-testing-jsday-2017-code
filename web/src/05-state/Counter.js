import React from 'react';

export class Counter extends React.Component {
  state = {
    value: 0
  };

  componentDidMount() {
    this.interval = setInterval(() => {
      this.setState({ value: this.state.value + 1 });
    }, 1000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    return <div>{this.state.value}</div>;
  }
}
