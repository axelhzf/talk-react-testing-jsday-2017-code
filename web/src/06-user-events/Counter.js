import React from 'react';

export class Counter extends React.Component {

  state = {
    value: 0
  }

  componentDidMount () {
    this.startInterval()
  }

  componentWillUnmount () {
    this.stopInterval();
  }

  resetInterval() {
    this.setState({ value: 0 });
    this.stopInterval();
    this.startInterval();
  }

  startInterval() {
    this.interval = setInterval(() => {
      this.setState({ value: this.state.value + 1 });
    }, 1000);
  }

  stopInterval() {
    if (this.interval) clearInterval(this.interval);
  }

  handleResetClick = () => this.resetInterval();

  handleKeyPress = (e) => {
    if (e.key === 'r') {
      this.resetInterval();
    }
  }

  render () {
    return (
      <div className="container" tabIndex={0} onKeyPress={this.handleKeyPress}>
        <div>{this.state.value}</div>
        <button onClick={this.handleResetClick}>Reset</button>
      </div>
    )
  }
}