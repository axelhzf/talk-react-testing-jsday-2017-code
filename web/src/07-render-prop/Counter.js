import React from 'react';
import PropTypes from 'prop-types';

export class Counter extends React.Component {
  static propTypes = {
    render: PropTypes.func
  };

  state = {
    value: 0
  };

  componentDidMount() {
    this.startInterval();
  }

  componentWillUnmount() {
    this.stopInterval();
  }

  resetInterval() {
    this.setState({ value: 0 });
    this.stopInterval();
    this.startInterval();
  }

  startInterval() {
    this.interval = setInterval(() => {
      this.setState({ value: this.state.value + 1 });
    }, 1000);
  }

  stopInterval() {
    if (this.interval) clearInterval(this.interval);
  }

  handleResetClick = () => this.resetInterval();

  render() {
    const { render } = this.props;
    return (
      <div>
        {render({
          value: this.state.value,
          reset: this.handleResetClick
        })}
      </div>
    );
  }
}
