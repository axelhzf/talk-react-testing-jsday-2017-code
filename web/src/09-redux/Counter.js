import React from 'react';
import { combineReducers, createStore } from 'redux';
import { connect } from 'react-redux';

function counterReducer(state = 0, action) {
  switch (action.type) {
    case 'INCREMENT':
      return state + 1;
    case 'DECREMENT':
      return state - 1;
    default:
      return state;
  }
}

export const reducer = combineReducers({
  counter: counterReducer
});

const increment = () => ({ type: 'INCREMENT' });
const decrement = () => ({ type: 'DECREMENT' });

export const createDefaultStore = () => {
  const store = createStore(
    reducer,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
  );
  return store;
};

function Counter({ value, increment, decrement }) {
  return (
    <div>
      <div>{value}</div>
      <button className="increment" onClick={increment}>
        +1
      </button>
      <button className="decrement" onClick={decrement}>
        -1
      </button>
    </div>
  );
}

const mapStateToProps = state => ({
  value: state.counter
});

export const CounterContainer = connect(mapStateToProps, {
  increment,
  decrement
})(Counter);
