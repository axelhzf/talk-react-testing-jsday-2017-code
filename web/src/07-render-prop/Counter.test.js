import React from 'react';
import { shallow } from 'enzyme';
import { Counter } from './Counter';
import _ from 'lodash';

jest.useFakeTimers();

describe('07-render-prop', () => {
  it('should render the initial value with 0', () => {
    const wrapper = shallow(
      <Counter render={({ value }) => <div>{value}</div>} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should reset the counter', () => {
    const wrapper = shallow(
      <Counter
        render={({ value, reset }) => (
          <div>
            <div>{value}</div>
            <button onClick={reset}>reset</button>
          </div>
        )}
      />
    );
    jest.runTimersToTime(5000);
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
    wrapper.find('button').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
