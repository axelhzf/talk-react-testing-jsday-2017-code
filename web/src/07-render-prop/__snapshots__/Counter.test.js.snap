// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`07-render-prop should render the initial value with 0 1`] = `
<div>
  <div>
    0
  </div>
</div>
`;

exports[`07-render-prop should reset the counter 1`] = `
<div>
  <div>
    <div>
      5
    </div>
    <button
      onClick={[Function]}
    >
      reset
    </button>
  </div>
</div>
`;

exports[`07-render-prop should reset the counter 2`] = `
<div>
  <div>
    <div>
      0
    </div>
    <button
      onClick={[Function]}
    >
      reset
    </button>
  </div>
</div>
`;
